shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/RuleServiceImpl.java

```diff
@@ -152,11 +152,17 @@ public int update(final RuleDTO ruleDTO) {
         Assert.notNull(before, "the updated rule is not found");
         RuleDO ruleDO = RuleDO.buildRuleDO(ruleDTO);
         final int ruleCount = ruleMapper.updateSelective(ruleDO);
+
+        // need old data for cleaning
+        final List<RuleConditionDO> beforeRuleCondition = ruleConditionMapper.selectByQuery(new RuleConditionQuery(ruleDO.getId()));
```

Comment:
update beforeRuleCondition ？  why？

Reply:
because the gateway uses the prefix tree to handle the cache, I can not clean up the history cache if Conditions change, so I pass the pre-modified Conditions to the past for precision cleaning.